core/src/main/scala/kafka/server/ControllerServer.scala

```diff
@@ -175,6 +174,11 @@ class ControllerServer(
           OptionalLong.empty()
         }
 
+        val maxIdleIntervalNs = config.metadataMaxIdleIntervalMs match {
```

Comment:
It's more traditional to have conversion stuff like this in KafkaConfig. If you had a function like `maxIdleIntervalNs` there, it could return Option[Long] and we could convert that into OptionalLong as needed (I think a simple asScala can do it if you have the optional converter imported...)

Reply:
Okay. I fixed this. `asJava` converts it to a `Optional[Long]` so I did it manually.